plugins/modules/alert_rule.py

```diff
@@ -0,0 +1,460 @@
+    api_password: "{{ api_password }}"
+    validate_certs: "{{ validate_certs }}"
+    state: present
+    name: alert_rule_new
+    condition: NEW
+    categories:
+      - all
+      - SYS_DISK_EVENTS
+      - NODE_STATUS_EVENTS
+    channels:
+      - SupportAssist
+    eventgroup_ids:
+      - 100010001
+      - 100010002
+    exclude_eventgroup_ids:
+      - 100010005
+    interval: 1100
+    transient: 10
+    limit: 10
+    severities:
+      - emergency
+      - critical
+
+- name: To delete the existing alert condition
+  dellemc.powerscale.alert_rule:
+    onefs_host: "{{ onefs_host }}"
+    api_user: "{{ api_user }}"
+    api_password: "{{ api_password }}"
+    validate_certs: "{{ validate_certs }}"
+    state: absent
+    name: alert_rule_new
+'''
+
+RETURN = r'''
+changed:
+  description: A boolean indicating if the task had to make changes.
+  returned: always
+  type: bool
+  sample: "false"
+alert_conditions:
+  description: The alert condition details.
+  type: dict
+  returned: always
+  contains:
+    name:
+      description: The name of the alert condition.
+      type: str
+    condition:
+      description: The condition of the alert condition.
+      type: str
+    categories:
+      description: The categories of the alert condition.
+      type: list
+      elements: str
+    channels:
+      description: The channels of the alert condition.
+      type: list
+      elements: str
+    eventgroup_ids:
+      description: The event group IDs of the alert condition.
+      type: list
+      elements: str
+    exclude_eventgroup_ids:
+      description: The event group categories of the alert condition.
+      type: list
+      elements: str
+    interval:
+      description: The interval of the alert condition.
+      type: int
+    limit:
+      description: The limit of the alert condition.
+      type: int
+    severities:
+      description: The severities of the alert condition.
+      type: list
+      elements: str
+    transient:
+      description: The transient of the alert condition.
+      type: int
+  sample:
+    {
+      "name": "alert_rule_new",
+      "condition": "NEW",
+      "categories": ["all", "SYS_DISK_EVENTS"],
+      "channels": ["SupportAssist"],
+      "eventgroup_ids": ["100010001", "100010002", "100010003"],
+      "exclude_eventgroup_ids": ["100010005"],
+      "interval": 11,
+      "limit": 10,
+      "severities": ["emergency"],
+      "transient": 10
+    }
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.dellemc.powerscale.plugins.module_utils.storage.dell.shared_library.powerscale_base \
+    import PowerScaleBase
+from ansible_collections.dellemc.powerscale.plugins.module_utils.storage.dell \
+    import utils
+from ansible_collections.dellemc.powerscale.plugins.module_utils.storage.dell.shared_library.events \
+    import Events
+from copy import deepcopy
+
+LOG = utils.get_logger('alert_rule')
+
+CATEGORY_DICT = {
```

Comment:
Can we use the API instead of hard coding the dict. ?
https://github.com/Isilon/isilon_sdk_python/blob/Isilon_SDK_v0.3.0.1/isilon_sdk/isilon_sdk/v9_5_0/docs/EventApi.md#get_event_categories

Reply:
Yes, updated